tox.ini

```diff
@@ -0,0 +1,47 @@
+[tox]
+minversion = 3.7.0
+envlist =
+    pre-commit-ci
+skipsdist = true
+
+[testenv]
+passenv = http_proxy HTTP_PROXY https_proxy HTTPS_PROXY no_proxy NO_PROXY
+usedevelop = true
+basepython = python3
+deps =
+  setuptools>=7.0
+
+[testenv:pre-commit-install]
+basepython = python3
+deps = pre-commit
+commands =
+    pre-commit install
+    pre-commit install --hook-type commit-msg
+
+[testenv:pre-commit-uninstall]
+basepython = python3
+deps = pre-commit
+commands =
+    pre-commit uninstall
+    pre-commit uninstall --hook-type commit-msg
+
+[testenv:pre-commit-autoupdate]
+basepython = python3
+deps = pre-commit
+commands =
+    pre-commit autoupdate
+
+[testenv:pre-commit]
+basepython = python3
+deps = pre-commit
+passenv = HOME
+commands =
+    pre-commit run --all-files
+
+[testenv:pre-commit-ci]
+basepython = python3
+deps = pre-commit
+passenv = HOME
+commands =
+    pre-commit run --all-files --show-diff-on-failure --from-ref a73d443c1d02698d9f3e030947c469677798bd32 --to-ref HEAD
```

Comment:
> What is it "a73d443c1d02698d9f3e030947c469677798bd32"?

It is the hash that references the commit just below this PR.
Only file modified since this commit will be linted by pre-commit.

$ git log -4 --oneline
53568f4ca (HEAD -> trailing_blanks, orangeopensource/trailing_blanks) [CI][doc] Add some documentation for tox and pre-commit
1b1341dec [CI] Trigger tox/pre-commit linter(s) in Azure CI
1aebc1646 [CI] Add a pre-commit trailing blanks linter via tox
a73d443c1 [CI][doc][build] Trim src folder files trailing blanks (#15162)